projects/baby_driver/scripts/can_util.py

```diff
@@ -183,3 +183,35 @@ def next_message(
         ))
 
     return msg
+
+def can_pack(data_list):
+    """
+    Converts list of tuples in form ((int) val, (int) len_in_bytes)
+            and combines them into a bytearray rendition
+    Returns:
+        A bytearray object representing message components passed in
+    Raises:
+        ValueError: if tuple has more than 2 items
+        ValueError: if tuple value exceeds allotted length in bytes
+    """
+    to_bytes = ""
+    for item in data_list:
+        if len(item) > 2:
+            raise ValueError("Tuple length {}: expected 2".format(len(item)))
+        #store value and number of bytes allotted
+        val = item[0]
+        len_in_bytes = item[1]
+        #check that values are valid
+        if len_in_bytes < 1 or val < 0:
+            raise ValueError("len in bytes must be > 0; val must be non-negative")
```

Comment:
You should document that val can't be negative in the docstring (probably just say it must be in `[0, 2**len_in_bytes - 1]`).

(It could be a possible improvement to support negative numbers in 2's complement as well!)